### PR TITLE
fix: can access sampled data for thickness calculations

### PR DESCRIPTION
**Pull request by rabii-chaarani**
A quick fix to access sampled contacts and sampled structural data for thickness calculations while I work on the proper solution. Can access sampled contacts from MapData

### CODE REVIEW

File: map2loop/interpolators.py [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+        pass
+
+    @beartype.beartype
+    @abstractmethod
+    def setup_grid(self, bounding_box: dict):
+        """
+        abstract method to setup an XY grid (abstract method)
+
+        Args:
+            bounding_box (dict): a dictionary containing the bounding box of the map data
```
Comment (AngRodrigues) [inline map2loop/interpolators.py]: should there be a specification for the structure of the dict bbox? something like: 

{
'xmin': 0, 
'ymin': 0, 
'xmax': 1, 
'ymax':1
}

File: map2loop/interpolators.py [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+        bounding_box: dict,
+        structure_data: pandas.DataFrame,
+        interpolator: Any = None,
+    ) -> Any:
+        """
+        Execute interpolate method (abstract method)
+
+        Args:
+            bounding_box (dict): a dictionary containing the bounding box of the map data
+            structure_data (pandas.DataFrame): sampled structural data
```
Comment (AngRodrigues) [inline map2loop/interpolators.py]: Maybe specify column headers for structural data?

**Review by AngRodrigues — APPROVED** [review]

File: map2loop/__init__.py
Comment (lachlangrose) [inline map2loop/__init__.py:1]: Why did you delete the __init__.py file? It is needed for exposing the objects we want the user to use in map2loop

**Review by lachlangrose — CHANGES_REQUESTED** [review]
Add the __init__.py file back please

File: map2loop/__init__.py
Comment (rabii-chaarani) [reply map2loop/__init__.py:1]: I didn't delete it, it's added in pull request  #48

**Comment by lachlangrose** [comment]
If we merge this pull request and don't merge #47 now then M2L won't work.
So it's better to keep each pr self contained it possible.

File: map2loop/__init__.py
Comment (rabii-chaarani) [reply map2loop/__init__.py:1]: I recovered the **init.py** file

File: map2loop/interpolators.py (hunk 1 above)
Comment (rabii-chaarani) [reply map2loop/interpolators.py]: docstring updated to include dictionary format enforced in mapdata.py

File: map2loop/interpolators.py (hunk 2 above)
Comment (rabii-chaarani) [reply map2loop/interpolators.py]: I don't think it's worth it since we are going to access sampled data directly from the samples storage class, and the argument structure_data will be removed
